### PR TITLE
Bundle autoupdate

### PR DESCRIPTION
**Pull request by janisslsm**
relies on Aliucord/AliucordNative#4

### CODE REVIEW

File: bootstrap/index.ts [hunk 1]
```diff
@@ -27,13 +28,22 @@
             }
         }
 
-        const aliucordDir = externalStorageDirectory + "/AliucordRN";
+        const aliucordDir = `${externalStorageDirectory}/AliucordRN`;
         AliuFS.mkdir(aliucordDir);
 
-        const bundlePath = aliucordDir + "/Aliucord.js.bundle";
-        if (!AliuFS.exists(bundlePath)) await download("https://raw.githubusercontent.com/Aliucord/AliucordRN/builds/Aliucord.js.bundle", bundlePath);
+        const commits = await fetch("https://api.github.com/repos/Aliucord/AliucordRN/commits?sha=builds");
```
Comment (TymanWasTaken) [inline bootstrap/index.ts:34]: You probably need to add error handling, if for example it gets ratelimited

**Review by TymanWasTaken — CHANGES_REQUESTED** [review]

File: bootstrap/index.ts (hunk 1 above)
Comment (rushiiMachine) [inline bootstrap/index.ts:34]: use the commit hash for the file itself + max limit 1

like this

https://github.com/Aliucord/AliucordManager/blob/4bb45b4ea711eec499d4da7424c96263ed930ca3/app/src/main/kotlin/com/aliucord/manager/network/service/AliucordGithubService.kt#L25-L27

File: bootstrap/index.ts [hunk 2]
```diff
@@ -27,13 +28,22 @@
...
+        const json = await commits.json();
+        const latestCommit = json[0].sha;
+
+        const externalBundlePath = `${aliucordDir}/Aliucord.js.bundle`;
+        const internalBundlePath = `${codeCacheDirectory}/${latestCommit}.js.bundle`;
```
Comment (rushiiMachine) [inline bootstrap/index.ts:39]: Aliucord.$hash.js.bundle

**Review by rushiiMachine — CHANGES_REQUESTED** [review]

File: bootstrap/index.ts (hunk 1 above)
Comment (rushiiMachine) [reply bootstrap/index.ts:34]: this way it doesn't update every single time something else in builds updates

File: bootstrap/index.ts [hunk 3]
```diff
@@ -31,16 +31,24 @@
...
-        const latestCommit = json[0].sha;
-
         const externalBundlePath = `${aliucordDir}/Aliucord.js.bundle`;
-        const internalBundlePath = `${codeCacheDirectory}/${latestCommit}.js.bundle`;
         if (AliuFS.exists(externalBundlePath)) {
             globalThis.aliucord = AliuHermes.run(externalBundlePath);
             return;
         }
+
+        const commits = await fetch("https://api.github.com/repos/Aliucord/AliucordRN/commits?sha=builds&path=bootstrap.js&per_page=1");
```
Comment (rushiiMachine) [inline bootstrap/index.ts]: not bootstrap.js

**Review by janisslsm — COMMENTED** [review]
Should it be checked if the file exists or always download the bundle if ratelimit is hit

File: bootstrap/index.ts [hunk 4]
```diff
@@ -31,16 +31,24 @@
...
+        if (commits.status === 403) {
+            const internalBundlePath = `${codeCacheDirectory}/Aliucord.js.bundle`;
+            if (!AliuFS.exists(internalBundlePath)) await download("https://raw.githubusercontent.com/Aliucord/AliucordRN/builds/Aliucord.js.bundle", internalBundlePath);
+            globalThis.aliucord = AliuHermes.run(internalBundlePath);
+
+            return;
+        }
+        const json = await commits.json();
```
Comment (rushiiMachine) [inline bootstrap/index.ts:48]: throw body as text if status != 200 before parsing to JSON

File: bootstrap/index.ts [hunk 5]
```diff
@@ -31,16 +31,24 @@
...
-        const internalBundlePath = `${codeCacheDirectory}/${latestCommit}.js.bundle`;
         if (AliuFS.exists(externalBundlePath)) {
             globalThis.aliucord = AliuHermes.run(externalBundlePath);
             return;
         }
+
+        const commits = await fetch("https://api.github.com/repos/Aliucord/AliucordRN/commits?sha=builds&path=bootstrap.js&per_page=1");
+        if (commits.status === 403) {
+            const internalBundlePath = `${codeCacheDirectory}/Aliucord.js.bundle`;
+            if (!AliuFS.exists(internalBundlePath)) await download("https://raw.githubusercontent.com/Aliucord/AliucordRN/builds/Aliucord.js.bundle", internalBundlePath);
```
Comment (rushiiMachine) [inline bootstrap/index.ts:43]: you'd need to download every time here cause therea no way to tell if it's old

**Review by rushiiMachine — CHANGES_REQUESTED** [review]

File: bootstrap/index.ts (hunk 5 above)
Comment (janisslsm) [reply bootstrap/index.ts:43]: yop, was going to ask.

File: bootstrap/index.ts (hunk 3 above)
Comment (janisslsm) [reply bootstrap/index.ts]: yeah, I fixed this in last commit

**Pull request merged**